### PR TITLE
Fixes #48 replace poster by requests

### LINKED ISSUE #48: Replace old poster lib by modern requests
[Requests](http://docs.python-requests.org/en/master/) is more mature and stable library, besides supporting a more rich set of features than [poster](https://atlee.ca/software/poster/). In addition, poster doesn't support Python3, so it would be nice to replace it.

### PR DESCRIPTION
**Pull request by eribeiro**

### CODE REVIEW

**Comment by eribeiro** [comment]
Hi @dvanaken, 

Whenever you have time, please let me know what you think of this PR. :smiley:

**Comment by dvanaken** [comment]
@eribeiro thanks @bohanjason will review this soon.

File: server/website/script/controller_simulator/upload_data.py [hunk 1]
```diff
@@ -25,20 +20,22 @@ def upload(basedir, upload_code):
...
+                                    files=params,
+                                    data={'upload_code':  upload_code})
+            print(response)
+
             sample_idx += 1
 
 if __name__ == "__main__":
-    if len(sys.argv) != 3:
-        print "Usage: python upload_data.py [datadir] [upload_code]"
+    if len(sys.argv) < 3:
```
Comment (bohanjason) [inline server/website/script/controller_simulator/upload_data.py]: len < 3 or len > 4

**Comment by bohanjason** [comment]
Comments are inline. It looks good. Could you please update the branch to resolve the conflicts ?

**Comment by eribeiro** [comment]
Hi @dvanaken and @bohanjason! Gonna solve the conflicts soon as well as address the review comments. Thanks!

File: server/website/script/controller_simulator/upload_data.py (hunk 1 above)
Comment (eribeiro) [reply server/website/script/controller_simulator/upload_data.py]: Cool. Gonna adress this.

**Comment by eribeiro** [comment]
@dvanaken Okay, I did a first pass to fix the conflicts. Let me know if I missed something. Thanks!

**Comment by eribeiro** [comment]
TravisCI is throwing the following error:
```
Running source validators:
  check_pylint
  check_pycodestyle
  check_illegal_patterns
  check_header
Traceback (most recent call last):
  File "script/validators/source_validator.py", line 370, in <module>
    main()
  File "script/validators/source_validator.py", line 355, in main
    status = validate_dir(target)
  File "script/validators/source_validator.py", line 169, in validate_dir
    if not validate_file(file_path):
  File "script/validators/source_validator.py", line 145, in validate_file
    file_path, validator.config_path)
  File "script/validators/source_validator.py", line 190, in check_pylint
    res.return_code, res.stderr)
AssertionError: return_code=1, expected=0
```
@dvanaken @bohanjason Any idea of what can be happening here?

**Comment by dvanaken** [comment]
@eribeiro Try running pylint (with our custom config file) on the files you changed.  Directions are on the wiki: https://github.com/cmu-db/ottertune/wiki/Style-&-Formatting#pylint. I'll fix the exception in the source validation script later.

**Comment by eribeiro** [comment]
@dvanaken Thanks for the heads up! :) I fixed the issue and now it's able to be merged. But I strongly advise @bohanjason to review again before merging, just in case. Thanks!

**Comment by eribeiro** [comment]
@bohanjason @dvanaken I merged this PR with the latest master branch head to keep it up to date.

**Pull request merged**